R/inject.R

```diff
@@ -0,0 +1,38 @@
+# Evaluates an expression (like `runApp()`) with the shiny.http.response.filter
+# option set to a function which rewrites the <head> to include shiny-tracer.js.
+with_shinytest_js <- function(expr) {
+  shiny::addResourcePath(
+    "shinytest",
+    system.file("js", package = "shinytest")
+  )
+
+  filter <- function(request, response) {
+    if (response$status < 200 || response$status > 300) return(response)
+
+    # Don't break responses that use httpuv's file-based bodies.
+    if ('file' %in% names(response$content))
+      return(response)
+
+    if (!grepl("^text/html\\b", response$content_type, perl=T))
+      return(response)
+
+    # HTML files served from static handler are raw. Convert to char so we
+    # can inject our head content.
+    if (is.raw(response$content))
+      response$content <- rawToChar(response$content)
+
+    # Modify the <head> to load shinytest.js
+    response$content <- sub(
+      "</head>",
+      "<script src=\"shinytest/shiny-tracer.js\"></script>\n</head>",
+      response$content,
+      ignore.case = TRUE
+    )
+
+    return(response)
+  }
+
+  oldOptions <- options(shiny.http.response.filter = filter)
```

Comment:
We don't care about the old values? Is that OK? I have no idea, just asking. 

If `shiny.http.response.filter` is a function, then we can create a function that is the sequence of the old value and the new `filter`, so that both run. 

But again, I don't know what you use `shiny.http.response.filter` for.